src/lib/workspace/Variable.tsx

```diff
@@ -31,32 +32,95 @@ export function VariableDetails(props: Props) {
     analysisState,
   } = props;
   const studyMetadata = useStudyMetadata();
+
+  // set showMore state
+  const [showMore, setShowMore] = useState(false);
+
+  // find the number of variables for multifilter case
+  const numberOfProviderLabel = MultiFilterVariable.is(variable)
+    ? findMultifilterVariableLeaves(
+        variable,
+        groupBy(entity.variables, (variable) => variable.parentId)
+      ).length
+    : 1;
+
+  // show the first three if multifilter variable
+  const threeProviderLabel = MultiFilterVariable.is(variable)
+    ? findMultifilterVariableLeaves(
+        variable,
+        groupBy(entity.variables, (variable) => variable.parentId)
+      ).map((variable, i) => {
```

Comment:
The linter points out: `Unnecessary escape character: \[`. Could you fix? (You can probably just replace `\]` with `]`.)

The linter also points out `Array.prototype.map() expects a value to be returned at the end of arrow function`. To fix that, I would return either `null` or  `<></>` when `i >= 3`.

Reply:
@jtlong3rd you are always welcome to bother me 😄 
Anyway, actually I knew that the first warning but forgot to address it. Yes it works by simply removing the backslash. As for the second, I added `return null;` for those cases

Reply:
Wouldn't it make more sense to only `map` the first three items in the array? It's fewer iterations and clearer, imo.

```typescript
findMultifilterVariableLeaves(
  variable,
  groupBy(entity.variables, (variable) => variable.parentId)
)
.slice(3)
.map((variable, i) => {
```

Reply:
I agree with @dmfalke here - you could `.slice(0, 3)` to map the first three elements.

Reply:
Thank you for the clarification @jtlong3rd~